doc/developers-guide/accumulators.md

```diff
@@ -3,25 +3,25 @@
 The stanza is processed and eventually it and/or other messages are sent out, either to the original sender, to another c2s process within the same MongooseIM installation, or to another XMPP server.
 
-At the beginning of the main processing chain an accumulator is created containing the original stanza, some values extracted from the stanza and additional info:
-
-* ref - a unique reference of the acc, useful for tracing
-* timestamp - current time
-* element - the original stanza
-* name - <<"message">>, <<"presence">> or <<"iq">>
-* type - e.g. <<"set">> (in iq)
-* attrs - attributs of the root xml element of the stanza
-* from, to - full jids of the sender and recipient in binary form
-* from_jid, to_jid - same as #jid{} records.
-* origin_sid - session id (SID) of `ejabberd_c2s`. Set when we receive a stanza
-  from the user device. *property*, *optional*.
-* origin_jid - full JID of `ejabberd_c2s`.
-  Set the same time as `origin_sid`. *property*, *optional*.
+At the beginning of the main processing chain an accumulator is created containing following set of keys:
+
+* `ref` - A unique reference of the acc, useful for tracing.
+* `timestamp` - An Erlang timestamp retrieved from `os:timestamp()`.
+* `origin_pid` - A PID of the process that created the accumulator.
+* `origin_location` - `{Module, Function Line}` - A place in the code where accumulator was created.
+* `origin_stanza` - Original stanza that triggered the processing (in a binary).
+* `lserver` - Nameprepped domain in which context the processing is being done.
+* `stanza` - A map with information about the stanza being routed. May be missing in some processing chains (when they are not triggered by a stanza)!
+  * `element` - `exml:element()` with the current stanza being routed.
+  * `from_jid`, `to_jid` - `jid:jid()` with the sender and the recipient.
+  * `name` - A name of top-level element in `element`.
+  * `type` - A value of `type` attribute of top-level element. If the attribute is missing, this field contains `undefined`.
+  * `ref` - A reference of routed stanza.
 
 It is then passed through all the stages until it reaches the end of its life.
 Throughout the process it is the very same accumulator; it is therefore possible to store a value in it on one stage of the processing and retrieve the same value later on.
 
-The main assumption is that whatever the MongooseIM does, it is always triggered by a stanza entering the system, one way or another (with an exeption for REST methods and mongooseimctl, that should have their own methods of creating and processing an accumulator).
+The main assumption is that whatever the MongooseIM does, it is always triggered by a stanza entering the system, with some exceptions, such as a couple of `mongooseimctl` operations, which create stanza-less accumulators.
```

Comment:
The main assumption is that whatever MongooseIM does, it is always triggered by a stanza entering the system, with some exceptions, such as a couple of `mongooseimctl` operations, which create stanza-less accumulators.